### DISCUSSION TITLE
Weight of Ragdoll Bones

### DISCUSSION THREAD

**Discussion by motherd** [jrouwe/JoltPhysics.js#199] [Q&A] [answered]
I have been utilizing the ragdoll feature and would like to ask how to specify the weight for each individual bone.  
When building the ragdoll, I tried setting the weight using `Jolt.RagdollPart.part.GetMassProperties().set_mMass` or `Jolt.Shape.GetMassProperties().set_mMass`, but it does not seem to be reflected.  
Could you please advise on the correct approach?

**Comment by jrouwe** [comment, marked as answer]
`GetMassProperties()` is a function that calculates the mass based on members of the `RagdollPart` structure and the volume of the shape. It returns a temporary, so you cannot change it.

Use `<ragdoll_part>.mOverrideMassProperties = EOverrideMassProperties_CalculateInertia` and `<ragdoll_part>.mMassPropertiesOverride.mMass = ...` instead.

**Reply by motherd** [reply]
I was able to set the mass. Thank you for your response.
